framework/connection.py

```diff
@@ -0,0 +1,78 @@
+"""
+Module to connect to remote host
+"""
+
+import logging
+
+from common_ci_utils.connection import Connection
+from framework import config
+from paramiko.auth_handler import AuthenticationException, SSHException
+
+log = logging.getLogger(__name__)
+
+
+class SSHConnection:
+    """
+    A class that connects to remote host
+    """
+
+    _instance = None
+
+    def __new__(cls, *args, **kwargs):
+        if not cls._instance:
+            cls._instance = super(SSHConnection, cls).__new__(cls)
+            cls._instance._conn = None
+        return cls._instance
+
+    def __init__(self):
+        """
+        Get connection to host
+
+        Raises:
+            authException: In-case of authentication failed
+            sshException: In-case of ssh connection failed
+
+        """
+        # Initialize the connection only if it hasn't been created yet
+        if not self._conn:
+            self.host = config.ENV_DATA["noobaa_sa_host"]
+            self.user = config.ENV_DATA["user"]
+            self.password = config.ENV_DATA.get("password")
+            self.private_key = config.ENV_DATA.get("private_key")
```

Comment:
```suggestion
            self.private_key = config.ENV_DATA.get("private_key")
            cls._instance._conn = None
```

Reply:
__init__ method should not take cls as a parameter

Reply:
As discussed offline, I meant to initiate directly in instance via self